app/models/user.rb

```diff
@@ -691,6 +715,17 @@ def generic?
     !(admin? || teacher? || editor?)
   end
 
+  # for lectures that are older than one year, only the teacher or an editor
+  # of the course it belongs to can view users in forms related to the lecture
+  def can_view_users?(lecture)
+    return false unless can_edit?(lecture)
+    return true if can_edit?(lecture.course) || lecture.teacher == self
+    return true if lecture.course.term_independent
+    return true if !lecture.older_than?(1.year)
```

Comment:
This condition `lecture.older_than?(1.year)` is used in multiple places, so it's probably a good idea to outsource it, e.g. write a short method `is_lecture_stale()`.